Add GSoC 2024 VulnTotal Browser Extension Project Report

michaelehab opened the pull request:
### Tasks

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://aboutcode.readthedocs.io/en/latest/doc_maintenance.html#continuous-integration) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

AyanSinhaMahapatra reviewed (changes requested):
@michaelehab looks great, specially the challenges part. Thanks!

1. Can you add a section like this: https://aboutcode--146.org.readthedocs.build/en/146/archive/gsoc/reports/2024/scancodeio_scorecode_pranay.html#project-reference-links
2. Can you fix the documentation tests, these are minor cosmetic adjustments

michaelehab commented:
Hello @AyanSinhaMahapatra 
Thanks for the review!
I added the official GSoC page, I already had the links to the proposal and the github repo.
I also fixed the tests, please check them again.
For the extension links, I guess we can add them later when the extension becomes available on a store, or if we can finish this soon I can add it as a part of this PR.

AyanSinhaMahapatra reviewed (approved):
Thanks @michaelehab , merging!
